### PR TITLE
Removes lag

### PR DESCRIPTION
**Pull request by ninjacha**
Ports /tg/s upgraded master controller and subsystems.

How this works:

The controller(s) will check if there are ticks left to use, and if so, will process there stuff.
If there isn't any ticks left to use (checked with world.tick_usage, a new 510 var) they will sleep and wait until there is and fire.

### CODE REVIEW

**Comment by ninjacha** [comment]
WEW REOPEN BECAUSE FUCK

**Comment by ninjacha** [comment]
Forgot to mention. RUN THIS UNDER 510.1327! IT WON'T WORK ON ANYTHING HIGHER UNTIL 1333 IS OUT

**Comment by octareenroon91** [comment]
1333 and 1334 are out. Please advise.

**Comment by ninjacha** [comment]
Tested, they work with this.

**Comment by Koriath** [comment]
Part of the HuD is gone (health doll, O2 and Hunger sign)
We might need to fix this.
In adition, visual effects like people getting flashed are also fucked.
They appear, but they don't fade out.

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -437,86 +437,54 @@
...
-					if(-80 to -70)
-						I.icon_state = "passage7"
-					if(-90 to -80)
-						I.icon_state = "passage8"
-					if(-95 to -90)
-						I.icon_state = "passage9"
-					if(-INFINITY to -95)
-						I.icon_state = "passage10"
-				damageoverlay.overlays += I
+/mob/living/carbon/handle_regular_hud_updates()
```
Comment (octareenroon91) [inline code/modules/mob/living/carbon/life.dm:440]: W E W
E
W

note line 436 above.

**Pull request merged**